homeassistant/components/proxmoxve/__init__.py

```diff
@@ -111,91 +115,100 @@ def build_client() -> ProxmoxAPI:
                 continue
             except SSLError:
                 _LOGGER.error(
-                    'Unable to verify proxmox server SSL. Try using "verify_ssl: false"'
+                    "Unable to verify proxmox server SSL. "
+                    'Try using "verify_ssl: false" for proxmox instance %s:%d',
+                    host,
+                    port,
                 )
                 continue
+            except ConnectTimeout:
+                _LOGGER.warning("Connection to host %s timed out during setup", host)
+                continue
+
+            hass.data[PROXMOX_CLIENTS][host] = proxmox_client
 
-            return proxmox_client
+    await hass.async_add_executor_job(build_client)
 
-    proxmox_client = await hass.async_add_executor_job(build_client)
+    coordinators = hass.data[DOMAIN][COORDINATORS] = {}
 
-    async def async_update_data() -> dict:
-        """Fetch data from API endpoint."""
+    # Create a coordinator for each vm/container
+    for host_config in config[DOMAIN]:
+        host_name = host_config["host"]
+        coordinators[host_name] = {}
+
+        proxmox_client = hass.data[PROXMOX_CLIENTS][host_name]
+
+        # Skip invalid hosts
+        if proxmox_client is None:
+            continue
 
         proxmox = proxmox_client.get_api_client()
```

Comment:
Is this async safe? No blocking or I/O?

Reply:
Yes anything calling the proxmox api library is wrapped in a function and called using 
`await hass.async_add_executor_job(function) `  for example. If you are specifically asking about line 125 in __init__.py that is simply a getter for the api client, and calling get_api_client() doesn't make any API calls.